data_ingestion.py

```diff
@@ -21,22 +24,30 @@ def configure_logging():
     return logging.getLogger("AutoGPT-Ingestion")
 
 
-def ingest_directory(directory: str, memory: VectorMemory, args):
+def ingest_directory(directory: str, memory: VectorMemory, config: Config):
     """
     Ingest all files in a directory by calling the ingest_file function for each file.
 
-    :param directory: The directory containing the files to ingest
-    :param memory: An object with an add() method to store the chunks in memory
+    Args:
+        directory: The directory containing the files to ingest
+        memory: The memory to ingest the files into
+        config: The Config object containing the workspace path
     """
     logger = logging.getLogger("AutoGPT-Ingestion")
     try:
-        files = list_files(directory)
+        files = list_files(directory, config)
         for file in files:
-            ingest_file(file, memory, args.max_length, args.overlap)
+            ingest_file(os.path.join(config.workspace_path, file), memory)
     except Exception as e:
         logger.error(f"Error while ingesting directory '{directory}': {str(e)}")
 
 
+def setup_workspace(cfg):
+    workspace_directory = Path(__file__).parent / "autogpt" / "auto_gpt_workspace"
```

Comment:
I don't know when this became the _de facto_ workspace path, but it doesn't seem right to me.

Also duplicates hard-coded default workspace path elsewhere, which is impractical. Maybe create a constant in the workspace module with the default path?

Reply:
Agree, reworked.